src/omit-deep.js

```diff
@@ -0,0 +1,59 @@
+import { omit, isNil } from 'ramda';
+import curry from './curry';
+
+/**
+ * Checks if a value is an object.
+ * @private
+ * @param {*} obj The value to check.
+ * @returns {boolean} Returns `true` if `obj` is an object, else `false`.
+ */
+function isObject(obj) {
+  return Object.prototype.toString.call(obj) === '[object Object]';
+}
+
+/**
```

Comment:
I suggest we change the jsdoc to something like:

```js
/**
 * Recursively omit the specified keys from an object.
 * The keys are omitted from the object itself and all its nested objects.
 *
 * @template T
 * @see https://github.com/jonschlinkert/omit-deep/tree/master
 * @example
 *  omitDeep(['b', 'd'], { a: 1, b: 2, c: { d: 3, e: 4 } }); // { a: 1, c: { e: 4 } }
 * @param {string[]} props The list of keys to omit.
 * @param {T} input The value to omit keys from.
 * @returns {T} A new value with the specified keys omitted.
 */
```

- The function seems to accept any kind of values and it always return the same value passed to it. So a `@template` would be useful here.
- The `@param` order looks to be inverted. `input` is the second argument to the function.
- Would be nice if the `@example` showed the removal of a deeply nested property.